specs/lightclient/validator.md

```diff
@@ -0,0 +1,198 @@
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [Custom types](#custom-types)
+- [Containers](#containers)
+  - [SyncCommitteeContribution](#synccommitteecontribution)
+- [Validator assignments](#validator-assignments)
+  - [Lookahead](#lookahead)
+- [Beacon chain responsibilities](#beacon-chain-responsibilities)
+  - [Block proposal](#block-proposal)
+    - [Preparing a `BeaconBlock`](#preparing-a-beaconblock)
+    - [Constructing the `BeaconBlockBody`](#constructing-the-beaconblockbody)
+      - [Sync committee](#sync-committee)
+    - [Packaging into a `SignedBeaconBlock`](#packaging-into-a-signedbeaconblock)
+  - [Attesting and attestation aggregation](#attesting-and-attestation-aggregation)
+  - [Sync committee contributions](#sync-committee-contributions)
+  - [Sync committee contribution aggregation](#sync-committee-contribution-aggregation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This document represents the expected behavior of an "honest validator" with respect to HF 1 of the Ethereum 2.0 protocol. It builds on the [previous document for the behavior of an "honest validator" from Phase 0](../phase0/validator.md) of the Ethereum 2.0 protocol. This previous document is referred to below as the "Phase 0 document".
+
+HF 1 introduces a new type of committee: the sync committee. Sync committees are responsible for signing each block of the canonical chain and there exists an efficient algorithm for light clients to sync the chain using the output of the sync committees. See the [sync protocol](./sync-protocol.md) for further details on the light client sync. Under this network upgrade, validators will need to track their participation in this new committee type and produce the relevant signatures as required. Block proposers will need to incorporate the (aggregated) sync committee signatures into each block they produce.
+
+## Prerequisites
+
+All terminology, constants, functions, and protocol mechanics defined in the [HF 1 -- The Beacon Chain](./beacon-chain.md) doc are requisite for this document and used throughout. Please see this document before continuing and use as a reference throughout.
+
+## Custom types
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `SyncCommitteeIndex` | `uint64` | Index of a validator within a sync committee |
+
+## Containers
+
+### SyncCommitteeContribution
+
+```python
+class SyncCommitteeContribution(Container):
+    # Slot during which this contribution is broadcast
+    slot: Slot
+    # Position of the validator within the sync committee
+    index: SyncCommitteeIndex
+    # Signature by the validator over the block root of the _previous_ slot
```

Comment:
Are we using the _previous_ slot opposed to the _current_ slot for better stability?

Reply:
that falls out as a side effect but i'd say the bigger reason is just that we use blockspace to include the sync committee's proof about a given block -- if we signed over the block root of the current slot and then included that data into the current block, it would change the current block root so we are back to square one.

Reply:
Ah i see. I thought it would have worked like attestation

slot N at 0: submit block (package attestations from N-1)
slot N at 1/3: submit attestations and sync committee objects

slot N+1 at 0:  submit block (package attestations and sync committee object from N)

So sync_committee object for current slot is ok because it's getting included in the next slot, this is similar to attestation handling today

Reply:
Ah, I think y'all are saying the same thing.

If a sync committee is assigned to slot `N`, they vote on what they see the block_root of slot `N` as. But this is then *included* in the block at slot `N+1`. So they vote on the current/assigned slot but when this makes it on chain it is a vote on the previous slot (same as an optimal attestation)

Reply:
So I am looking at the object now and it seems that `slot` and the slot of `block` that is being signed over should be the same, whereas the comments imply otherwise. Essentially, at start of slot `block` is broadcast. Halfway through slot, sync committee votes. Next slot (`slot+1`) next block producer includes any sync committee votes from previous slot

Reply:
right, the comments are confusing -- i will update

Reply:
Yeah, let's do that. It's how we assign `slot` to `Attestation` so fits our mental model already